packages/common/http/src/headers.ts

```diff
@@ -14,8 +14,8 @@ interface Update {
 
 /**
  * Represents the header configuration options for an HTTP request.
- *
- * Instances should be assumed immutable with lazy parsing.
+ * Instances are immutable with lazy parsing.
+ * Modifying methods return a cloned instance with the change.
```

Comment:
Could we add somehow that the original object is not changed?

```suggestion
 * Modifying methods return a cloned instance with the change, the original object is never changed.
```